### PR TITLE
Various keycast-log-mode improvements

### PR DESCRIPTION
**Pull request by minad**
Hey Jonas,

I improved the keycast-log-frame parameters and tried to isolate the keycast frame and buffer a little bit better. I hope you like these changes! Some of the settings are a bit opinionated maybe. Please let me know if you'd like to see some things changed or only take the commits which you agree with.

### CODE REVIEW

File: keycast.el [hunk 1]
```diff
@@ -99,8 +99,8 @@ with no argument and acts on `selected-window'.
   :group 'keycast
   :type 'function)
 
-(defcustom mode-line-keycast-format "%s%k%c%r"
-  "The format spec used by `mode-line-keycast'.
+(defcustom keycast-format-mode-line-format "%s%k%c%r"
+  "The format spec used by `keycast-mode-line'.
```
Comment (tarsius) [inline keycast.el:103]: I've dropped the first `format`.

File: keycast.el [hunk 2]
```diff
@@ -290,16 +290,16 @@ instead."
         (unless cons
           (setq keycast-mode nil)
           (user-error
-           "Cannot turn on %s.  %s not found in %s.  Try customizing %s."
+           "Cannot turn on %s.  %s not found in %s.  Try customizing %s"
```
Comment (tarsius) [inline keycast.el:293]: Generally I follow the convention of not ending messages with a period, but it feels weird for multi-sentence message and leaves me wondering if the message was somehow truncated. So I don't do it in those cases.

File: keycast.el (hunk 1 above)
Comment (minad) [reply keycast.el:103]: Oh yes, pushed.

File: keycast.el [hunk 3]
```diff
@@ -62,7 +62,7 @@ into `mode-line-format' after the element specified here."
   "Whether enabling `keycast-mode' removes elements to the right.
 
 When this is non-nil, then enabling `keycast-mode' not only
-inserts `mode-line-keycast' into `mode-line-format' but also
+inserts `keycast-mode-line' into `mode-line-format' but also
```
Comment (tarsius) [inline keycast.el:65]: Fine by me, would do it like that nowadays too. I am under the impression that in recent years there was a shift from "prefix symbols with the package name, except..." to "always prefix all symbols with the package name".

File: keycast.el (hunk 2 above)
Comment (minad) [reply keycast.el:293]: I don't now either. I usually just follow the conventions, so far without bad side effects. If the message is truncate the trailing period is so or so removed?

File: keycast.el (hunk 3 above)
Comment (minad) [reply keycast.el:65]: I've not come across any cases where an exception of the strict naming convention makes sense. Therefore I would go with the renaming here. We can also deprecate the old symbols for backward compatibility. So far I've not added deprecations.

**Comment by tarsius** [comment]
Thanks!

File: keycast.el (hunk 2 above)
Comment (tarsius) [reply keycast.el]: I mean the non-truncated message that lacks the period looks like it might be truncated (even though it is not).

Comment (minad) [reply keycast.el]: Hmm, right. I usually try to avoid writing multiple sentences to mitigate this problem ;)

File: keycast.el (hunk 3 above)
Comment (tarsius) [reply keycast.el:65]: I though that putting `define-*` first made sense and if I remember correctly that also was necessary to automatically `defun`-like styling.

Comment (minad) [reply keycast.el:65]: > I though that putting define- first made sense and if I remember correctly that also was necessary to automatically get defun-like styling.

Ah right. That's an argument. But I think you can still get the styling by adjusting some variables. So in the greater scheme of things I prefer if the naming convention is followed strictly. Of course this is just my preference, but package-lint is also pushing me into that direction :)

Comment (tarsius) [reply keycast.el:65]: > We can also deprecate the old symbols for backward compatibility.

For `mode-line-keycast-format` that would probably make sense, but then the namespace pollution would stick around. Shouldn't be too hard to notice that something has changed and how to fix it, so I think it is fine to not define the aliases in this case.

**Pull request merged**